File: testprojects/python_main/pipelines/PYTHON_UNITY_CATALOG/code/python_unity_catalog/graph/Subgraph_1/Reformat_3.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from python_unity_catalog.config.ConfigStore import *
from python_unity_catalog.udfs.UDFs import *

def Reformat_3(spark: SparkSession, in0: DataFrame) -> DataFrame:
    return in0.select(
        col("firstname"), 
        col("middlename"), 
        col("lastname"), 
        col("id"), 
        col("salary"), 
        col("processed"), 
        col("dob"), 
        col("weight"), 
        col("state"), 
        col("city"), 
        col("gender")
    )

File: testprojects/scala_main/pipelines/LIVY_SCALA/code/src/main/scala/io/prophecy/pipelines/livy_scala/graph/Reformat_3.scala
package io.prophecy.pipelines.livy_scala.graph

import io.prophecy.libs._
import io.prophecy.pipelines.livy_scala.config.ConfigStore._
import io.prophecy.pipelines.livy_scala.config.Context
import io.prophecy.pipelines.livy_scala.udfs.UDFs._
import io.prophecy.pipelines.livy_scala.udfs._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object Reformat_3 {

  def apply(context: Context, in: DataFrame): DataFrame =
    in.select(
      array(col("year"),                 col("value"), col("unit")).as("arr_str1"),
      array(col("industry_code_ANZSIC"), col("industry_name_ANZSIC"))
        .as("arr_str2"),
      array(lit(1), lit(2), col("year").cast(IntegerType)).as("arr_int"),
      struct(array(col("year"),                 col("value"), col("unit")).as("col1"),
             array(col("industry_code_ANZSIC"), col("industry_name_ANZSIC"))
               .as("col2"),
             col("unit")
      ).as("struct_complex"),
      col("year"),
      col("variable")
    )

}
